src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs

```diff
@@ -1426,9 +1426,24 @@ public void Redraw()
                 this.Redraw(this.ConnectorModel.End.Center);
             }
 
+            this.SetCollapsedByNodeViewModel();
             RaisePropertyChanged(nameof(ZIndex));
         }
 
+        /// <summary>
+        /// Evaluates whether both nodes associated with a connector are collapsed, if so, collapses the connector itself.
+        /// This is to address DYN-4449.  Connectors are only recorded in the Undo stack when they are connected.
+        /// Consequently, if a group is collapsed and then moved, performing an Undo operation will not restore
+        /// the connector to its state at the time the move was recorded.
+        /// </summary>
+        private void SetCollapsedByNodeViewModel()
```

Comment:
it seems like you could write a unit test for the Redraw method that makes sure IsCollapsed is true when it should be.

Reply:
please see the test included. hope that's okay.